simphony_mayavi/core/cuba_data.py

```diff
@@ -236,3 +244,48 @@ def _array_value(self, name, values):
         else:
             # The array is a CUBA attribute array.
             return values.get(cuba, self._defaults[cuba])
+
+
+def fix_attribute_arrays(attribute_data):
+    """ Fix the layout of the vtk attribute array container.
+
+    The function check that the container has only CUBA related arrays
+    and that they all have the same length. Any masks that are missing
+    are added to the container.
+
+    """
+    data = attribute_data
+
+    # check invalid array names
+    names = [
+        data.get_array_name(array_id)
+        for array_id in range(data.number_of_arrays)]
+    try:
+        cubas = [CUBA[name] for name in names if '-mask' not in name]
+    except KeyError:
+        raise ValueError("vtk object contains non cuba named arrays")
+
+    # check array length
+    lengths = {
+        len(data.get_array(array_id))
+        for array_id in range(data.number_of_arrays)}
+    if len(lengths) > 1:
+        info = {
+            data.get_array_name(array_id): len(
+                data.get_array_name(array_id))
+            for array_id in range(data.number_of_arrays)}
+        message = "vtk object arrays are not the same size: {}"
+        raise ValueError(message.format(info))
+    elif len(lengths) == 0:
+        # empty vtk object
+        return
+    else:
+        # fix masked arrays
+        length = next(iter(lengths))
```

Comment:
`next(iter(lenghts))` to get the item 1 of lengths is not fun to read. I'm not sure what to suggest there.`lenghts.pop()` ?